deepchecks/tabular/utils/feature_importance.py

```diff
@@ -309,6 +309,11 @@ def _calc_permutation_importance(
         elif not skip_messages:
             get_logger().info('Calculating permutation feature importance. Expected to finish in %s seconds',
                               predicted_time_to_run)
+    elif timeout == 0 and not skip_messages:
```

Comment:
I'm not sure I understood the purpose of this change. User is supposed to set timeout == 0 in case they want to avoid calculating feature importance entirely, which should be possible.  Only thing we should take care of as far as I understand is that a correct error message / warning is shown in case that this happens. 

What is the issue you where trying to avoid by adding this code? What error was raised when timeout is set to 0?

Reply:
The current logic means that if timeout is 0, then the task will always fail, raising a TimeOutError with the message that feature_importance would take too much time. This is the case since the variable predicted_time_to_run is at least 1.

However, this would confuse the user because the tutorials suggest that feature_importance can be skipped. Indeed, who would set timeout=0 and expect that it is indeed used to compare expected runtime?
Instead, I added the correct error message that timeout=0 is not implemented yet.


The alternative is to pass a "None" as feature_importance if timeout=0. In that case, if the model is not a Pipeline, then the downstream function might successfully extract a built-in feature importance (e.g. if model is a tree).
If that fails, however, the error will be nondescript ValueError with a similar confusion for the model - they request not to compute feature importance, and the error is "Was not able to calculate feature importance".

Bottom line for me: This functionality is not implemented right now, hence my choice.

I can do it either way, just let me know.

Reply:
Somehow I missed the exception handling in calculate_feature_importance_or_none. Now it makes sense :D 

Well please have a look if I missed anything. Now both None, 0, and an integer timeout should hopefully work...